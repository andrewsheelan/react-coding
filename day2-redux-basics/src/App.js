import React from 'react';
import Page from './Page';
import { Provider } from 'react-redux';
import store from './store';

function App() {
  return (
    <div className="App">
      <Provider store={store}>
        <Page />
      </Provider>
    </div>
  );
}

export default App;
