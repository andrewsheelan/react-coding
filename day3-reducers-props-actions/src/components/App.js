import React from 'react';
import { Provider } from 'react-redux';
import Header from './Header';
import store from '../store';

function App() {
  return (
    <div className="root">
      <Provider store={store}>
        <Header/>
      </Provider>
    </div>
  );
}

export default App; 
