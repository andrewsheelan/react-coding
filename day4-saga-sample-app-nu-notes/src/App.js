import React from "react";
import Header from "./components/nav/Header";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import Dashboard from "./components/home/Dashboard";
import Login from "./components/auth/Login";
import Signup from "./components/auth/Signup";
import NotesShow from "./components/notes/Show";
import NotesCreate from "./components/notes/Create";
import NotesIndex from "./components/notes/Index";
import { connect } from "react-redux";
function App({ user }) {
  return (
    <div className="App">
      <BrowserRouter>
        <Header />
        <Switch>
          <Route exact path="/" component={Dashboard} />
          <Route path="/login" component={Login} />
          <Route path="/signup" component={Signup} />
          <Route exact path="/notes/create" component={NotesCreate} />{" "}
          <Route path="/notes/:id" component={NotesShow} />
          <Route path="/notes" component={NotesIndex} />
          <Route path="*" component={Dashboard} />
        </Switch>
      </BrowserRouter>
    </div>
  );
}

const mapStateToProps = ({ auth: { user } }) => ({
  user,
});
export default connect(mapStateToProps)(App);
