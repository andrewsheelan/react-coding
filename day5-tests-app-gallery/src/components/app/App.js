import React from "react";
import Gallery from "../gallery/Gallery";
import "./App.scss";

function App() {
  return (
    <div className="app" data-testid="app">
      <Gallery />
    </div>
  );
}

export default App;
