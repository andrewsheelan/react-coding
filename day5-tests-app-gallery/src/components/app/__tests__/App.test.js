import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../App";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"; //ES6 modules

const middlewares = [];
const mockStore = configureStore(middlewares);
const initialState = {
  gallery: {
    loading: false,
    errors: {
      show: false,
      message: "",
    },
    data: {
      photos: [],
    },
  },
};
const store = mockStore(initialState);

it("renders correctly", () => {
  const { queryByTestId, } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(queryByTestId("app")).toBeTruthy();

  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
