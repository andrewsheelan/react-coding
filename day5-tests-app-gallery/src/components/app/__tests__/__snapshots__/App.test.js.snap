// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
  data-testid="app"
>
  <div
    className="gallery"
  >
    <div
      className="gallery-error error"
    >
      
    </div>
    <div
      className="gallery-loading"
    >
      
    </div>
    <div
      className="gallery-photos"
    />
  </div>
</div>
`;
